spec/models/king_spec.rb

```diff
@@ -94,4 +94,74 @@
       expect(moved_king.moved).to eq true
     end
   end
+
+  describe 'castling' do
+    let(:castling_king) do
+      empty_game.pieces.create(
+        type: 'King',
+        color: 'white',
+        x_position: 4,
+        y_position: 7
+      )
+    end
+    let(:castling_rook) do
+      empty_game.pieces.create(
+        type: 'Rook',
+        color: 'white',
+        x_position: 7,
+        y_position: 7
+      )
+    end
+    let(:obstruction) do
+      empty_game.pieces.create(
+        type: 'Bishop',
+        color: 'white',
+        x_position: 1,
+        y_position: 7
+      )
+    end
+    let(:obstructed_rook) do
+      empty_game.pieces.create(
+        type: 'Rook',
+        color: 'white',
+        x_position: 0,
+        y_position: 7
+      )
+    end
+
+    it 'should return true on valid castling' do
+      castling_rook.save
+      expect(castling_king.move!(6, 7)).to eq true
```

Comment:
You should try to limit to one `expect` per `it`. Otherwise, one condition failing means that the rest of the suite will not run and you will not have a clear picture of what is actually broken.